ResearchKit/Common/ORKGraphView.m

```diff
@@ -0,0 +1,848 @@
+    }];
+}
+
+- (void)updateScrubberLineAccessories:(CGFloat)xPosition {
+    CGFloat scrubberYPos = [self canvasYPointForXPosition:xPosition];
+    CGFloat scrubbingVal = [self valueForCanvasXPosition:(xPosition)];
+   [self.scrubberThumbView setCenter:CGPointMake(xPosition + ORKGraphLeftPadding, scrubberYPos + ORKGraphTopPadding)];
+    self.scrubberLabel.text = [NSString stringWithFormat:@"%.0f", scrubbingVal];
+   CGSize textSize = [self.scrubberLabel.text boundingRectWithSize:CGSizeMake(320, CGRectGetHeight(self.scrubberLabel.bounds))
+                                                           options:(NSStringDrawingUsesFontLeading|NSStringDrawingUsesLineFragmentOrigin)
+                                                        attributes:@{NSFontAttributeName:self.scrubberLabel.font}
+                                                           context:nil].size;
+   [self.scrubberLabel setFrame:CGRectMake(CGRectGetMaxX(self.scrubberLine.frame) + 6, CGRectGetMinY(self.scrubberLine.frame), textSize.width + 8, CGRectGetHeight(self.scrubberLabel.frame))];
+}
+
+- (CGFloat)snappedXPosition:(CGFloat)xPosition {
+    CGFloat widthBetweenPoints = CGRectGetWidth(self.plotsView.frame) / self.xAxisPoints.count;
+    NSUInteger positionIndex;
+    for (positionIndex = 0; positionIndex < self.xAxisPoints.count; positionIndex++) {
+        
+        CGFloat dataPointValue = ((ORKRangePoint *)self.dataPoints[positionIndex]).maximumValue;
+        
+        if (dataPointValue != NSNotFound) {
+            CGFloat value = [self.xAxisPoints[positionIndex] floatValue];
+            
+            if (fabs(value - xPosition) < (widthBetweenPoints * ORKGraphSnappingClosenessFactor)) {
+                xPosition = value;
+            }
+        }
+    }
+    return xPosition;
+}
+
+- (CGFloat)valueForCanvasXPosition:(CGFloat)xPosition {
+    BOOL snapped = [self.xAxisPoints containsObject:@(xPosition)];
+    CGFloat value = NSNotFound;
+    NSUInteger positionIndex = 0;
+    if (snapped) {
+        for (positionIndex = 0; positionIndex<self.xAxisPoints.count-1; positionIndex++) {
+            CGFloat xAxisPointVal = [self.xAxisPoints[positionIndex] floatValue];
+            if (xAxisPointVal == xPosition) {
+                break;
+            }
+        }
+        value = ((ORKRangePoint *)self.dataPoints[positionIndex]).maximumValue;
+    }
+    return value;
+}
+
+- (void)setScrubberViewsHidden:(BOOL)hidden animated:(BOOL)animated {
+    if ([self numberOfValidValues] > 0) {
+        CGFloat alpha = hidden ? 0 : 1;
+        
+        if (animated) {
+            [UIView animateWithDuration:0.2 animations:^{
+                self.scrubberThumbView.alpha = alpha;
+                self.scrubberLine.alpha = alpha;
+                self.scrubberLabel.alpha = alpha;
+            }];
+        } else {
+            self.scrubberThumbView.alpha = alpha;
+            self.scrubberLine.alpha = alpha;
+            self.scrubberLabel.alpha = alpha;
```

Comment:
To reduce code duplication, personally I prefer declaring the block as a local variable; then pass in the block as a variable in `if {}`, and fire the block in `else {}`.